### PR TITLE
feat: Docker - repo list input convert to tags

### PR DESCRIPTION
**Pull request by RohitRaj011**
# Description

Docker Repo - List Input will be converted to tags when user presses enter/tab/space or separates it via comma.

Fixes https://github.com/devtron-labs/devtron/issues/5234

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/dockerRegistry/Docker.tsx [hunk 1]
```diff
@@ -1444,6 +1446,59 @@ const DockerForm = ({
...
+                        value: [
+                            ...prev.repositoryList.value,
+                            {
+                                label: customState.repositoryList.inputValue,
+                                value: customState.repositoryList.inputValue,
+                            },
+                        ],
+                    },
+                }))
+                event.preventDefault()
```
Comment (eshankvaish) [inline src/components/dockerRegistry/Docker.tsx:1483]: - We should set the trimmed value in the chip
- Create a common handler for updating the state that can be reused in blur and change

File: src/components/dockerRegistry/Docker.tsx [hunk 2]
```diff
@@ -1452,15 +1507,28 @@ const DockerForm = ({
...
+                        components={CreatableComponents}
+                        onChange={(value) => {
+                            setCustomState((prev) => ({ ...prev, repositoryList: { ...prev.repositoryList, value } }))
+                        }}
+                        onInputChange={(inputValue) => {
+                            setCustomState((prev) => ({
+                                ...prev,
+                                repositoryList: { ...prev.repositoryList, inputValue },
+                            }))
+                        }}
```
Comment (eshankvaish) [inline src/components/dockerRegistry/Docker.tsx:1528]: let's refrain from using inline functions

**Review by eshankvaish — CHANGES_REQUESTED** [review]

File: src/components/dockerRegistry/Docker.tsx (hunk 1 above)
Comment (RohitRaj011) [reply src/components/dockerRegistry/Docker.tsx:1483]: trim won't be required because as soon as tab or space is pressed a tag is created

**Review by eshankvaish — DISMISSED** [review]

File: src/components/dockerRegistry/Docker.tsx [hunk 3]
```diff
@@ -1444,6 +1454,70 @@ const DockerForm = ({
         return renderRegistryCredentialsAutoInjectToClustersComponent()
     }
 
+    // CREATABLE METHODS
+    /**
+     * Sets the provided value in the repository list state.
+     * @param value value to set in the repository list state.
+     */
+    const setRepoListValue = (value: string) => {
+        setCustomState((prev) => ({
```
Comment (vivek-devtron) [inline src/components/dockerRegistry/Docker.tsx:1463]: We can directly set the state without callback

File: src/components/dockerRegistry/Docker.tsx [hunk 4]
```diff
@@ -1452,15 +1526,21 @@ const DockerForm = ({
...
                 <div className="mb-12">
                     <div className="dc__required-field fs-13 cn-9 mb-6">List of repositories</div>
-                    <textarea
-                        className="form__textarea"
-                        name="repositoryList"
+                    <CreatableSelect
+                        isMulti
+                        isClearable
                         autoFocus
                         value={customState.repositoryList?.value}
```
Comment (vivek-devtron) [inline src/components/dockerRegistry/Docker.tsx:1533]: Please check if this optional check is required?

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**